examples/atlas-api-key/create-and-assign-pak-together/versions.tf

```diff
@@ -2,7 +2,7 @@ terraform {
   required_providers {
     mongodbatlas = {
       source  = "mongodb/mongodbatlas"
-      version = "~> 1.0"
+      version = "~> 1.13.2"
```

Comment:
the downside of this is that when you merge this change this version won't be there. Are we ok with that? Another (longer) approach is that we leave the example as it is (by still setting the project_id) and once this is released to TF then we can update the example (or create a different one). LEt me know if I am missing something

Reply:
This is a good point which I had not contemplated. I do see this would apply to several scenarios where we modify examples, such as adding new attributes.
Users do have the possibility of seeing the examples section under a specific release tag, but understand this is probably not the regular flow.
One alternative I see to maintain examples compatible with latest release (in master branch) is to have new development in a separate branch (example `develop`), and merge into master during release process (this obviously has the downside of an additional step in our release process).

Reply:
Are you sure you need to update the version here? We have a [piece of logic](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/86df112a260ab00b9bbd188f1819ae9b23ad9eb7/.github/workflows/examples.yml#L31) in the action which overrides the version in the `version.tf` to be 99999 (which is the "current" built version)  so it should have your changes 


````
 name: go build
        run: go build -o terraform-plugin-dir/terraform-provider-mongodbatlas-dev_v99.99.99_x5 .
      - name: override plugin
        run: |
          # For newer versions
          mkdir -p ~/.terraform.d/plugins/registry.terraform.io/hashicorp/mongodbatlas-dev/99.99.99/"$(go env GOOS)"_"$(go env GOARCH)"/
          cp terraform-plugin-dir/terraform-provider-mongodbatlas-dev_v99.99.99_x5 ~/.terraform.d/plugins/registry.terraform.io/hashicorp/mongodbatlas-dev/99.99.99/"$(go env GOOS)"_"$(go env GOARCH)"/

````

Reply:
> Are you sure you need to update the version here?

There is no need from CI side, this change is more for users to know that this example (which does not define root project_id) can be used from 1.13.2 onwards. @marcosuma does raise a valid point that when this is merged, our master branch will contain examples which could cause confusion for users as it mentions a version that has not yet been released.

Reply:
> master branch will contain examples which could cause confusion for users as it mentions a version that has not yet been released

it makes sense. We can create a ticket to add automation (action) to update the provider version for all the examples to a specified version before the release. In this case, you won't need to update the version here and the new version will be in master only between pre-release and final release. This will fix another issue where we have examples in our repo with older provider versions and we should make sure that our examples are relevant with the latest release.

it's worth noting that we haven't received any issues or help requests regarding examples in the repository using an unreleased version so far. (saying to help triage this piece of work if we agree on creating the ticket)

Reply:
> ticket to add automation (action) to update the provider version for all the examples

I believe the main issue is when the content of the example changes and is **only** compatible with a version that will be released in the future, causing potential confusion for users that see examples from master branch. I do agree I don't see it as priority issue, and cannot see a simple solution either.

Reply:
I don't have major concerns here. I prefer to unblock you and resolve this one if we haven't seen a similar issue before. Feel free to resolve if you all are ok.

Reply:
Also @andreaangiolillo's suggestion is ok to me as well